### ISSUE TITLE
We may need a validator for iptables rules

### ISSUE THREAD

**Issue by gmiu** [HubbleStack/Nova#102]
When `firewall.py` checks for an incorrect written iptables rule, `iptables.check` returns `False`. This might mislead the user. The firewall plugin should specify somewhere that that rule had an incorrect syntax.

**Comment by basepi** [comment]
Is there already a way to validate these rules? An existing library or something? I feel like validating iptables syntax is a huge task!

**Comment by gmiu** [comment]
```
# salt-call iptables.check rule='-p tcp --dport 21 -m state --state ESTABLISHED, RELATED -j ACCEPT' chain=INPUT
[INFO    ] Executing command '/sbin/iptables -t filter -N 0x50569144dfL' in directory '/root'
[INFO    ] Executing command '/sbin/iptables -t filter -A 0x50569144dfL -p tcp --dport 21 -m state --state ESTABLISHED, RELATED -j ACCEPT' in directory '/root'
[ERROR   ] Command '/sbin/iptables -t filter -A 0x50569144dfL -p tcp --dport 21 -m state --state ESTABLISHED, RELATED -j ACCEPT' failed with return code: 2
[ERROR   ] output: iptables v1.4.7: "--state" requires a list of states with no spaces, e.g. ESTABLISHED,RELATED
Try `iptables -h' or 'iptables --help' for more information.
[INFO    ] Executing command '/sbin/iptables-save' in directory '/root'
[INFO    ] Executing command '/sbin/iptables -t filter -F 0x50569144dfL' in directory '/root'
[INFO    ] Executing command '/sbin/iptables -t filter -X 0x50569144dfL' in directory '/root'
local:
    False
```

**Comment by gmiu** [comment]
@basepi, `iptables.check` lets the user know when the syntax is wrong. But I'm not sure yet how to pick that error message inside the `audit` function. Also, I found some iptables library for python but I haven't looked at it yet.

**Comment by basepi** [comment]
Yeah, unfortunately that Error log won't be accessible from the calling context. You may have to write a custom version of iptables.check which returns error information, rather than just False.

**Comment by gmiu** [comment]
Fixed in #114. It's not a validator for iptables rule, just a new interpretation of the results from `iptables.check`. Closing this for now.
